### PR TITLE
Fix for dangling ports

### PR DESCRIPTION
**Pull request by bamsumit**
Issue Number: 246

Objective of pull request:

## Pull request checklist

Your PR fulfills the following requirements:
- [x] [Issue](https://github.com/lava-nc/lava/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [x] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [ ] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [x] Lint (`flakeheaven lint src/lava tests/`) and (`bandit -r src/lava/.`) pass locally
- [x] Build tests (`pytest`) passes locally

## Pull request type

 

 

Please check your PR type:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe): 

## What is the current behavior?

- Dangling input/output port does not work.

## What is the new behavior?

- They work

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by mathisrichter — CHANGES_REQUESTED** [review]
Looks good, Sumit! It may be worth adding a dedicated unit test for this though to make sure this capability does not break in the future.

**Comment by bamsumit** [comment]
> Looks good, Sumit! It may be worth adding a dedicated unit test for this though to make sure this capability does not break in the future.

Yeah I was looking at the best place to add that. I am not convinced it should go in `test_process.py` though. Suggestions?

File: tests/lava/magma/runtime/test_io_ports.py [hunk 1]
```diff
@@ -60,6 +62,8 @@ def __init__(self, proc):
         # the nested process
         self.p1 = P1()
         self.p1.out.connect(proc.out_ports.h_out)
+        # Reference h_var with var of the nested process
```
Comment (mathisrichter) [inline tests/lava/magma/runtime/test_io_ports.py:65]: Missing full stop

File: tests/lava/magma/runtime/test_io_ports.py [hunk 2]
```diff
@@ -252,6 +258,24 @@ def test_branching_hierarchical(self):
...
+        input not connected at all."""
+        receiver = HP2()
+        receiver.run(condition=RunSteps(num_steps=2),
+                     run_cfg=Loihi1SimCfg(select_sub_proc_model=True))
+        self.assertTrue(np.all(receiver.h_var.get() == np.array([0, 0])))
+        receiver.stop()
+
+    def test_dangling_output(self):
+        """Checks if the hierarchical process works with dangling output i.e.
+        output not conected at all."""
```
Comment (mathisrichter) [inline tests/lava/magma/runtime/test_io_ports.py:272]: conected -> connected

**Review by mathisrichter — APPROVED** [review]
Nice! I think the unit test is fine there. Since it is a feature of the compiler, you could also have written it as a part of the Compiler tests and only tested the contents of the builders, for instance. But this is also nice, then we have a working example. :)

**Review by PhilippPlank — APPROVED** [review]

**Review by mgkwill — APPROVED** [review]

**Pull request merged**